### ISSUE TITLE
Find nice administrative boarders

### ISSUE THREAD

**Issue by 00Bock** [00Bock/RubyCars#24]
maybe we can use this data:
http://biogeo.ucdavis.edu/data/gadm2.8/shp/MEX_adm_shp.zip

**Comment by kjoscha** [comment]
Before finding appropriate data, we should reflect whether 
1. which detail level of administrative units is absolutely necessary
2. capturing is possible via geocoding after all (after one initial run, geocoding is only necessary for new stations. Anyway we should be careful during development - yesterday i ran the whole scraping process for several times with geocoding)

**Comment by 00Bock** [comment]
It will be hard to generalise rules on administrative boarders, since their character differs from country to country... I guess anything smaller than ~5km diameter possibly makes little sense. I guess important criteria is that the names have a strong recognition (e.g. London and not Westminster). For Germany Level 3 Boarders (Gemeinden) make sense.

Geocoding makes sence when there is a table for stations with a specific id, so the process is not repeated with the usual scraping task but rather every day/week or so or when a new station is identified.

**Comment by kjoscha** [comment]
> Geocoding makes sence when there is a table for stations with a specific id, so the process is not repeated with the usual scraping task but rather every day/week or so or when a new station is identified.

we have this table, so it is not very dificult to implement it this way. Anyway an own geocoding service would make the system less fragile. At least we should take a deeper look on the request limitations of the external geocoding API's and base our decisions on these.

**Comment by kjoscha** [comment]
Bing unterstützt Batch-Geocoding :) Wir könnten also einen extra-Geocoding-Task einrichten, der für mit einem Request alle fehlenden Addressen sammelt.
https://developer.here.com/rest-apis/documentation/batch-geocoder/topics/request-submit.html

**Comment by kjoscha** [comment]
Quatsch, meinte natürliche Here (nicht Bing) und sehe gerade, dass es kostenpflichtig ist. Ich suche weiter. Aber wie gesagt, vielleicht reichen die Request-Limits auch, wenn wir nur fehlende Addressen updaten...

**Comment by 00Bock** [comment]
I'm just realising: there are always a lot of stars and no dots when setting up the environment at a new computer. if the requests get very big that might lead to request limits...

**Comment by BenSto** [comment]
Also ich glaube das Geocoding ist schon am Limit, gerade eben wollte ich kurz zipcar Barcelona hier auf dem Festrechner scrapen und Ergebnis war:

Mechanize::ResponseCodeError: 404 => Net::HTTPNotFound for http://nominatim.openstreetmap.org/reverse?format=json&lat=41.397434&lon=2.121495 -- unhandled response

**Comment by 00Bock** [comment]
Hier (http://gadm.org/version2) gibt's die ganze welt:
http://biogeo.ucdavis.edu/data/gadm2.8/gadm28_levels.shp.zip
